core/shared/src/main/scala/cats/effect/SyncIO.scala

```diff
@@ -497,6 +498,13 @@ object SyncIO extends SyncIOCompanionPlatform with SyncIOLowPriorityImplicits {
   val realTime: SyncIO[FiniteDuration] =
     RealTime
 
+  /**
+   * Creates a new instance of SecureRandom in a SyncIO context. This is cryptographically
+   * secure and thread-safe.
+   */
+  implicit lazy val secureRandom: SecureRandom[SyncIO] =
+    SecureRandom.javaSecuritySecureRandom[SyncIO].unsafeRunSync()
```

Comment:
You can do it exactly the same as the `IO` instance:

```suggestion
    SecureRandom.unsafeJavaSecuritySecureRandom[SyncIO]()
```